astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    @pytest.mark.parametrize('ignore', [False, True])
+    @pytest.mark.parametrize('fileobj', [False, True])
+    def test_no_resource_warning_raised_on_non_fits_file(self, ignore, fileobj):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/6168
+
+        The ResourceWarning shows up when (in python 3+) you try to
+        open a non-FITS file when using a filename.
+        """
+
+        filename = self.temp('not-fits.fits')
```

Comment:
@pllim - in one of my original commits I wrote a test - using the file name `'not-fits.fits'` - and made sure that it failed before I changed any code. You can see this at

https://github.com/DougBurke/astropy/blob/693a032cd0e5a13c4e10b8c6f60907252bedf8f3/astropy/io/fits/tests/test_hdulist.py#L918

It was not tested against the current `master` branch, but whatever was `master` at the time.